pkg/notifiers/slack/handle_request.go

```diff
@@ -189,7 +220,20 @@ func (n *Notifier) HandleRequestEvent(ctx context.Context, log *zap.SugaredLogge
 	return nil
 }
 
-func (n *Notifier) UpdateSlackMessage(ctx context.Context, slackClient *slack.Client, log *zap.SugaredLogger, rev access.Reviewer, req access.Request, rule rule.AccessRule, dbRequestor *identity.User) error {
+type UpdateSlackMessageOpts struct {
+	Ctx           context.Context
+	SlackClient   *slack.Client
+	Log           *zap.SugaredLogger
+	Review        access.Reviewer
+	Request       access.Request
+	RequestReview *access.Review
+	Rule          rule.AccessRule
+	DbRequestor   *identity.User
+}
+
+func (n *Notifier) UpdateSlackMessage(opts UpdateSlackMessageOpts) error {
+
+	ctx, slackClient, log, rev, req, reqRev, rule, dbRequestor := opts.Ctx, opts.SlackClient, opts.Log, opts.Review, opts.Request, opts.RequestReview, opts.Rule, opts.DbRequestor
```

Comment:
Where possible, avoiding this type of "destructuring" is preferred
In this case it looks like you have added this to avoid updating the existing function.

My preference would be updating the function to use opts.RequestReview.something etc rather than reqRev as its harder to read where the values come from